api/test_runs.go

```diff
@@ -130,3 +140,26 @@ func LoadTestRunsForFilters(ctx context.Context, filters shared.TestRunFilter) (
 	}
 	return shared.LoadTestRunsByKeys(ctx, keys)
 }
+
+func getPRCommits(ctx context.Context, pr int) []string {
+	log := shared.GetLogger(ctx)
+	secret, err := shared.GetSecret(ctx, "github-api-token")
+	if err != nil {
+		log.Debugf("Failed to load github-api-token: %s", err.Error())
+		return nil
+	}
+	oauthClient := oauth2.NewClient(ctx, oauth2.StaticTokenSource(&oauth2.Token{
+		AccessToken: secret,
+	}))
+	githubClient := github.NewClient(oauthClient)
+	commits, _, err := githubClient.PullRequests.ListCommits(ctx, "web-platform-tests", "wpt", pr, nil)
+	if err != nil || commits == nil {
+		log.Errorf("Failed to fetch PR #%v: %s", pr, err.Error())
+		return nil
+	}
+	shas := make([]string, len(commits))
+	for i := range commits {
+		shas[i] = commits[i].GetSHA()
```

Comment:
Here I wondered https://www.reddit.com/r/golang/comments/1m25a1/map_reduce_and_filter_in_go/, guess not :)